server/service/src/sync/translation_and_integration.rs

```diff
@@ -135,20 +135,31 @@ impl<'a> TranslationAndIntegration<'a> {
 
 impl IntegrationRecords {
     pub(crate) fn integrate(&self, connection: &StorageConnection) -> Result<(), RepositoryError> {
+        // Only start nested transaction is transaction is already ongoing. See integrate_and_translate_sync_buffer
+        let start_nested_transaction = { connection.transaction_level.get() > 0 };
```

Comment:
am I reading this correctly?
- initialised: create outer transaction and sub transaction for every upsert and every delete
- not initialised: no transactions at all

Reply:
Yep, hopefully this makes it clear: 6482b67cbc082a64849b77f6d98b856e87dc5aae